### DISCUSSION TITLE
Privacy plugin being sponsors-only is a bit problematic WRT Google Fonts usage

### DISCUSSION THREAD

**Discussion by ThiefMaster** [squidfunk/mkdocs-material#5679] [General]
As useful as Google Fonts it, it's a privacy nightmare. From what I understood the only way to get those fonts included locally in the build is using the privacy plugin - but you have to pay for that one.

Is it really a good idea to force people to pay or feed data to Google? I completely understand that you want to encourage people to sponsor the project, and I think if you are a commercial company using it, then doing so is the right thing to do anyway. But not giving Google more data than they already have seems like the kind of honorable thing that should be available to everyone...

I found [this suggestion](https://github.com/squidfunk/mkdocs-material/discussions/5475#discussioncomment-5835590) which might be better than using Google Fonts, but of course having the fonts embedded in the build is still the best option, also to avoid an extra dependency on an external host...

**Comment by squidfunk** [comment]
Great question, thanks for asking! From version 1.x, Material for MkDocs [makes it actually quite easy to disable Google Fonts](https://squidfunk.github.io/mkdocs-material/setup/changing-the-fonts/#autoloading):

``` yaml
theme:
  font: false
```

This will completely remove Google Fonts, thus not send any data to Google at all. Material for MkDocs will fall back to system fonts. If you want to use a custom font, you can do so very easily, [as our documentation describes](https://squidfunk.github.io/mkdocs-material/setup/changing-the-fonts/#additional-fonts).

This topic [has been discussed extensively in the past](https://github.com/squidfunk/mkdocs-material/issues/739), and [bundling all fonts with the theme is just not an option](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#+privacy.assets_exclude), as it will slow down _every_ build. This is the very raison d'être of the privacy plugin – your users' browser will make no requests to Google Fonts, as they are downloaded for self-hosting during build time. The privacy plugin is the simplest and hassle-free solution I could come up with, retaining the flexibility to use all available Google Fonts, while also completely respecting the users' privacy.

In fact, the privacy plugin was developed as a response to users raising concerns about Google Fonts, and it took us significant time to get it right. However, and I can't stress it enough: self-hosting or disabling are always an option. If you're tight on time and want a solution that just works, the privacy plugin might be a good choice. Additionally, I haven't checked, but there may be third party MkDocs plugins achieving the same thing.

Additional recommended reads:

- https://github.com/squidfunk/mkdocs-material/issues/739#issuecomment-1027713955
- https://github.com/squidfunk/mkdocs-material/issues/739#issuecomment-1036027153

> Is it really a good idea to force people to pay or feed data to Google? I completely understand that you want to encourage people to sponsor the project, and I think if you are a commercial company using it, then doing so is the right thing to do anyway. But not giving Google more data than they already have seems like the kind of honorable thing that should be available to everyone...

Nobody forces you to use Google Fonts. Since Material for MkDocs is based on Material Design, using Roboto as a default is obviously the most natural choice. However, as said, disabling Google Fonts is 1 line of configuration in `mkdocs.yml`. 

> I found https://github.com/squidfunk/mkdocs-material/discussions/5475#discussioncomment-5835590 which might be better than using Google Fonts, but of course having the fonts embedded in the build is still the best option, also to avoid an extra dependency on an external host...

It's another external service that you don't control. IMHO, downloading and automatic self-hosting is the canonical solution here, because it means that regardless of which font service or CDN you choose, you can be sure that you'll serve assets from your own servers. The privacy plugin will be released once the $14k funding goal is hit, which we expect to be the case by the end of this year. Then, it will be available to all users.

Please understand that releasing it earlier will counteract our Sponsorware strategy, and reduce the time of us maintainers to work on this project. Open Source is hard. Making Open Source sustainable is even harder.

**Reply by ThiefMaster** [reply]
Thanks for the pointer on  how to easily disable Google Fonts! Will look into this for sure.

And yeah I can fully understand your point WRT sponsorware. It's just a huge PITA when you work in academia to actually get such things paid - of course 15 bucks are nothing, but the red tape you have to go through.... well, there's a reason people sometimes pay things out of their own pocket just to avoid the hassle of getting reimbursed / going through their purchasing depts ;)

**Reply by ThiefMaster** [reply]
Thanks for pointing to the font customization link. In the end I went for this route and just downloaded the CSS and WOFF2 fonts from Google Fonts and put those in my own docs repo: https://github.com/indico/indico-user-docs/commit/c4413064252df22d35f774a5f9c50b66cbbc22b6

Maybe that commit is also useful for others who want to do the same thing. If yes, the easiest way is to clone my repo, and copy the `docs/stylesheets` folder into your own documentation repo and then just adapt mkdocs.yml to disable Google Fonts and load the custom stylesheet.

**Reply by squidfunk** [reply]
Thanks for sharing!
